File: java-client/src/main/java/org/openeclass/HelloBetaCMSWorld.java
/*
 * Copyright (C) 2005-2009 BetaCONCEPT LP.
 *
 * This file is part of BetaCMS.
 *
 * BetaCMS is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * BetaCMS is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with BetaCMS.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.openeclass;

import java.util.Calendar;

import org.betaconceptframework.betacms.repository.api.model.CalendarProperty;
import org.betaconceptframework.betacms.repository.api.model.ContentObject;
import org.betaconceptframework.betacms.repository.api.model.RepositoryUser;
import org.betaconceptframework.betacms.repository.api.model.StringProperty;
import org.betaconceptframework.betacms.repository.api.model.query.CacheRegion;
import org.betaconceptframework.betacms.repository.api.model.query.CmsOutcome;
import org.betaconceptframework.betacms.repository.api.model.query.CmsRankedOutcome;
import org.betaconceptframework.betacms.repository.api.model.query.criteria.ContentObjectCriteria;
import org.betaconceptframework.betacms.repository.api.security.BetaCmsCredentials;
import org.betaconceptframework.betacms.repository.client.BetaCmsRepositoryClient;
import org.betaconceptframework.betacms.repository.model.factory.CmsCriteriaFactory;

public class HelloBetaCMSWorld {
	
	public static void main(String[] args) throws Exception {

		BetaCmsRepositoryClient betaCmsRepositoryClient = new BetaCmsRepositoryClient("localhost");
		BetaCmsCredentials credentials = new BetaCmsCredentials("xxx", "yyy".toCharArray());

		//List<CmsRepository> availableRepositories = betaCmsRepositoryClient.getRepositoryService().getAvailableCmsRepositories();
		
		betaCmsRepositoryClient.login("myrepo", credentials);
		System.out.println("login OK");

		
		//Create a new content object instance of type 'genericContentResourceObject' 
		ContentObject helloWorldContentObject = betaCmsRepositoryClient.getCmsRepositoryEntityFactory()
												.newContentObjectForType("genericContentResourceObject", "el");
		
		//Provide owner for content object. Select system repository user
		RepositoryUser owner = betaCmsRepositoryClient.getRepositoryUserService().getSystemRepositoryUser();
		helloWorldContentObject.setOwner(owner);
		
		//Fill content object's properties with values
		
		//Profile properties
		//Title
		StringProperty profileTitleProperty = (StringProperty)helloWorldContentObject.getCmsProperty("profile.title");
		profileTitleProperty.setSimpleTypeValue("Hello βCMS world !");
		
		//Language
		StringProperty profileLanguageProperty = (StringProperty)helloWorldContentObject.getCmsProperty("profile.language");
		profileLanguageProperty.setSimpleTypeValue("el");
		
		//Last Modification Date
		CalendarProperty profileModifiedProperty = (CalendarProperty)helloWorldContentObject.getCmsProperty("profile.modified");
		profileModifiedProperty.setSimpleTypeValue(Calendar.getInstance());

		//Accessibility properties
		StringProperty canBeReadByProperty = (StringProperty)helloWorldContentObject.getCmsProperty("accessibility.canBeReadBy");
		canBeReadByProperty.addSimpleTypeValue("ALL");
		
		StringProperty canBeUpdatedByProperty = (StringProperty)helloWorldContentObject.getCmsProperty("accessibility.canBeUpdatedBy");
		canBeUpdatedByProperty.addSimpleTypeValue("NONE");
		
		StringProperty canBeDeletedByProperty = (StringProperty)helloWorldContentObject.getCmsProperty("accessibility.canBeDeletedBy");
		canBeDeletedByProperty.addSimpleTypeValue("NONE");
		
		StringProperty canBeTaggedByProperty = (StringProperty)helloWorldContentObject.getCmsProperty("accessibility.canBeTaggedBy");
		canBeTaggedByProperty.addSimpleTypeValue("ALL");
		

		// body
		String body = "<b>My first βCMS content object</b>";
		StringProperty bodyProperty = (StringProperty)helloWorldContentObject.getCmsProperty("body");
		bodyProperty.setSimpleTypeValue(body);
		
		// save content object
		helloWorldContentObject = betaCmsRepositoryClient.getContentService().saveContentObject(helloWorldContentObject, false);
		System.out.println("Successfully saved content object");
		
		// retrieve content object
		helloWorldContentObject = betaCmsRepositoryClient.getContentService().getContentObjectById(helloWorldContentObject.getId(), CacheRegion.FIVE_MINUTES);
		System.out.println("Successfully retrieved content object");
		System.out.println(helloWorldContentObject.toXml());
		
		//betaCmsRepositoryClient.getContentService().deleteContentObject(helloWorldContentObject.getId());
		//System.out.println("Successfully deleted content object");
		
		//owner.getSpace().addContentObjectReference(helloWorldContentObject.getId());
		//System.out.println("Successfully added content object to owner's space");
		//betaCmsRepositoryClient.getSpaceService().getOrganizationSpace().addContentObjectReference(helloWorldContentObject.getId());
		//System.out.println("Successfully added content object to organization space");
		
		
		
		//Search content object
		//Create search criteria
		ContentObjectCriteria contentObjectCriteria = CmsCriteriaFactory.newContentObjectCriteria();
		contentObjectCriteria.addFullTextSearchCriterion("Hello βCMS world");
		
		//You can cache results
		contentObjectCriteria.setCacheable(CacheRegion.FIVE_MINUTES);
		
		//Execute query
		CmsOutcome<CmsRankedOutcome<ContentObject>> cmsOutcome = betaCmsRepositoryClient.getContentService().searchContentObjects(contentObjectCriteria);
		
		//Iterate through results
		for (CmsRankedOutcome<ContentObject> cmsRankedOutcome : cmsOutcome.getResults()) {
			ContentObject contentObject = cmsRankedOutcome.getCmsRepositoryEntity();
			
			StringProperty titleProperty = (StringProperty)contentObject.getCmsProperty("profile.title");
			String contentObjectType = contentObject.getContentObjectType();
			
			System.out.println("Found content object of type "+contentObjectType+" with title "+titleProperty.getSimpleTypeValue());
			
			// delete every entry
			//betaCmsRepositoryClient.getContentService().deleteContentObject(contentObject.getId());
		}
	}

}

File: php-examples/hellobetacmsworld.php
<?php
/*========================================================================
*   Open eClass 2.1
*   E-learning and Course Management System
* ========================================================================
*  Copyright(c) 2003-2009  Greek Universities Network - GUnet
*
*  Developers Group: Thanos Kyritsis
*
*  Open eClass is an open platform distributed in the hope that it will
*  be useful (without any warranty), under the terms of the GNU (General
*  Public License) as published by the Free Software Foundation.
*
*  Contact address: 	GUnet Asynchronous eLearning Group,
*  			Network Operations Center, University of Athens,
*  			Panepistimiopolis Ilissia, 15784, Athens, Greece
*  			eMail: info@openeclass.org
* =========================================================================*/ 
?>
<?php
echo "<pre>";
require_once("http://localhost:8080/JavaBridgeTemplate554/java/Java.inc");

$betacmsClient = new Java("org.betaconceptframework.betacms.repository.client.BetaCmsRepositoryClient", "localhost");

//Retrieve a list of all available repositories managed by connected BetaCMS Server
$availableRepositories = java_values($betacmsClient->getRepositoryService()->getAvailableCmsRepositories());
// echo a list of these available repository ids
foreach ($availableRepositories as $key => $cmsRepository) {
	echo "repo".$key.": " .$cmsRepository->getId() ."\n";
}
        
//Retrieve information about BetaCMS repository with id 'demorepo'
$betacmsRepository = $betacmsClient->getRepositoryService()->getCmsRepository("myrepo");
echo "We retrieved information about BetaCMS repository with id: " .$betacmsRepository->getId() ."\n";

//Check if BetaCMS repository with id "demorepo" is available in connected BetaCMS Server
if (java_values($betacmsClient->getRepositoryService()->isRepositoryAvailable("myrepo"))) {
	echo "demorepo is available\n";
}

//Create user credentials
$pass = new Java("java.lang.String", "yyy");
$passCh = $pass->toCharArray();
$credentials = new Java("org.betaconceptframework.betacms.repository.api.security.BetaCmsCredentials", "xxx", $passCh);
        
//Login to repository with ID 'demorepo'
$betacmsClient->login("myrepo", $credentials);


//Retrieve built in Subject Taxonomy
$subjectTaxonomy = $betacmsClient->getTaxonomyService()->getBuiltInSubjectTaxonomy("en");

//Change label for English Locale and save change to repository
$subjectTaxonomy->addLocalizedLabel("en", "BetaCMS Subject Taxonomy");
$betacmsClient->getTaxonomyService()->saveTaxonomy($subjectTaxonomy);

//Search Taxonomy
$myTaxonomy = $betacmsClient->getTaxonomyService()->getTaxonomy("myTaxonomy", "en");

//Delete taxonomy
if (!java_is_null($myTaxonomy)) {
	$betacmsClient->getTaxonomyService()->deleteTaxonomyTree($myTaxonomy->getId());
}


//Create a new content object instance of type 'genericContentResourceObject'
$newContentObject = $betacmsClient->getCmsRepositoryEntityFactory()->newContentObjectForType("genericContentResourceObject", "en");


//Provide owner for content object. Select system repository user
$owner = $betacmsClient->getRepositoryUserService()->getSystemRepositoryUser();
$newContentObject->setOwner($owner);

//Provide a human readable name for new content object
$newContentObject->setSystemName("myNewContentObject");


//Profile properties
//Title
$profileTitleProperty = $newContentObject->getCmsProperty("profile.title");
$profileTitleProperty->setSimpleTypeValue("Title for new ContentOBject");
        
//Language
$profileLanguageProperty = $newContentObject->getCmsProperty("profile.language");
$profileLanguageProperty->setSimpleTypeValue("en");
        
//Accessibility properties
//Notice the method used to add value : addSimpleTypeValue
//This is because property 'accessibility.canBeReadBy' can contain multiple values
$canBeReadByProperty = $newContentObject->getCmsProperty("accessibility.canBeReadBy");
$canBeReadByProperty->addSimpleTypeValue("ALL");
        
$canBeUpdatedByProperty = $newContentObject->getCmsProperty("accessibility.canBeUpdatedBy");
$canBeUpdatedByProperty->addSimpleTypeValue("NONE");
        
$canBeDeletedByProperty = $newContentObject->getCmsProperty("accessibility.canBeDeletedBy");
$canBeDeletedByProperty->addSimpleTypeValue("NONE");
        
$canBeTaggedByProperty = $newContentObject->getCmsProperty("accessibility.canBeTaggedBy");
$canBeTaggedByProperty->addSimpleTypeValue("ALL");


$body = "<b>Body for new  content object</b>";
$bodyProperty = $newContentObject->getCmsProperty("body");
$bodyProperty->setSimpleTypeValue($body);


//Use ContentService to save new instance
//ALWAYS get a reference to method result as new instance is 
//populated with entity identifier. Properties profile.created and profile.modified
//are automatically created as well.
$newContentObject = $betacmsClient->getContentService()->saveContentObject($newContentObject, false);


//You can retrieve content object using its id
$newContentObject = $betacmsClient->getContentService()->getContentObjectById($newContentObject->getId(), java("org.betaconceptframework.betacms.repository.api.model.query.CacheRegion")->FIVE_MINUTES);

//Export ContentObject to XML
//If variable is not re-assigned xml export will reflect 
//unsaved content object status
echo $newContentObject->toXml();


echo "</pre>";
?>
